DOCSUP-12413: macros support in functions cluster and clusterAllReplicas (#27759)
ClickHouse/ClickHouse: docs/en/sql-reference/table-functions/cluster.md

# cluster, clusterAllReplicas {#cluster-clusterallreplicas}

Allows to access all shards in an existing cluster which configured in `remote_servers` section without creating a [Distributed](../../engines/table-engines/special/distributed.md) table. One replica of each shard is queried.

`clusterAllReplicas` function — same as `cluster`, but all replicas are queried. Each replica in a cluster is used as a separate shard/connection.

!!! note "Note"
    All available clusters are listed in the [system.clusters](../../operations/system-tables/clusters.md) table.

**Syntax**

``` sql
cluster('cluster_name', db.table[, sharding_key])
cluster('cluster_name', db, table[, sharding_key])
clusterAllReplicas('cluster_name', db.table[, sharding_key])
clusterAllReplicas('cluster_name', db, table[, sharding_key])
```
**Arguments**

- `cluster_name` – Name of a cluster that is used to build a set of addresses and connection parameters to remote and local servers. 
- `db.table` or `db`, `table` - Name of a database and a table.  

> - `sharding_key` -  A sharding key if the cluster has more than one shard. Optional.

Author: ```suggestion
- `sharding_key` -  A sharding key. Optional. Needs to be specified if the cluster has more than one shard. 
```

> - `sharding_key` - When insert into cluster function with more than one shard, sharding key needs to be provided. Optional.

Reviewer: ```suggestion
- `sharding_key` -  A sharding key if the cluster has more than one shard. Optional.
```

**Returned value**

The dataset from clusters.

**Using Macros**

> Arguments can contain macros — substitutions in curly brackets. The substituted values are taken from the [macros](../../operations/server-configuration-parameters/settings.md#macros) section of the server configuration file.

Author: ```suggestion
`cluster_name` can contain macros — substitution in curly brackets. The substituted value is taken from the [macros](../../operations/server-configuration-parameters/settings.md#macros) section of the server configuration file.
```
